backend/src/packages/trucks/truck.controller.ts

```diff
@@ -0,0 +1,375 @@
+ *         - $ref: '#/components/schemas/ErrorType'
+ *         - type: object
+ *           properties:
+ *             message:
+ *               type: string
+ *               enum:
+ *                 - Truck already exists!
+ * securitySchemes:
+ *   bearerAuth:
+ *     type: http
+ *     scheme: bearer
+ *     bearerFormat: JWT
+ */
+class TruckController extends Controller {
+  private truckService: TruckService;
+
+  public constructor(logger: ILogger, truckService: TruckService) {
+    super(logger, ApiPath.TRUCKS);
+
+    this.truckService = truckService;
+
+    this.addRoute({
+      path: TruckApiPath.ROOT,
+      method: 'POST',
+      validation: {
+        body: truckRequestBodyValidationSchema,
+      },
+      handler: (request) =>
+        this.create(
+          request as ApiHandlerOptions<{
+            body: TruckEntityT;
+          }>,
+        ),
+    });
+
+    this.addRoute({
+      path: TruckApiPath.$ID,
+      method: 'PUT',
+      validation: {
+        body: truckUpdateRequestBodyValidationSchema,
+        params: truckIdParameterValidationSchema,
+      },
+      handler: (request) =>
+        this.update(
+          request as ApiHandlerOptions<{
+            body: Partial<TruckEntityT>;
+            params: { id: number };
+          }>,
+        ),
+    });
+
+    this.addRoute({
+      path: TruckApiPath.ROOT,
+      method: 'GET',
+      handler: () => this.getAll(),
+    });
+
+    this.addRoute({
+      path: TruckApiPath.$ID,
+      method: 'GET',
+      validation: {
+        params: truckIdParameterValidationSchema,
+      },
+      handler: (request) =>
+        this.get(
+          request as ApiHandlerOptions<{
+            params: { id: number };
+          }>,
+        ),
+    });
+
+    this.addRoute({
+      path: TruckApiPath.$ID,
+      method: 'DELETE',
+      validation: {
+        params: truckIdParameterValidationSchema,
+      },
+      handler: (request) =>
+        this.delete(
+          request as ApiHandlerOptions<{
+            params: { id: number };
+          }>,
+        ),
+    });
+  }
+
+  /**
+   * @swagger
+   * /trucks/:
+   *   post:
+   *     summary: Create a new truck
+   *     tags:
+   *       - truck
+   *     requestBody:
+   *       description: Truck data to be added
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             $ref: '#/components/schemas/Truck'
+   *     responses:
+   *       '201':
+   *         description: Truck created successfully
+   *         content:
+   *           application/json:
+   *             schema:
+   *               $ref: '#/components/schemas/TruckResponse'
+   *       '400':
+   *         description: Bad request
+   *
+   */
+  private async create(
+    options: ApiHandlerOptions<{
+      body: TruckEntityT;
+    }>,
+  ): Promise<ApiHandlerResponse> {
+    return {
+      status: HttpCode.CREATED,
+      payload: await this.truckService.create(options.body),
```

Comment:
TruckEntityT includes 'id', but 'id' shouldn't be in body when we creating a truck

Reply:
corrected